src/strategies/clqdr-balance-with-lp/index.ts

```diff
@@ -0,0 +1,139 @@
+  'function getVirtualSupply() external view returns (uint256)',
+]
+
+const bn = (num: any): BigNumber => {
+  return BigNumber.from(num.toString());
+};
+
+const addUserBalance = (userBalances, user: string, balance) => {
+  if (userBalances[user]) {
+    return (userBalances[user] = userBalances[user].add(balance));
+  } else {
+    return (userBalances[user] = balance);
+  }
+}
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+
+  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
+
+  const clqdrTotalRes = await call(provider, contractAbi, [
+    beetsVaultAddress,
+    'getPoolTokens',
+    [clqdrPoolId]
+  ]);
+
+  const totalClqdrInBeets = bn(clqdrTotalRes[1][1]);
+
+  const virtualSupplyRes = await call(provider, contractAbi, [
+    lpAddress,
+    'getVirtualSupply',
+    []
+  ]);
+
+  const virtualSupply = bn(virtualSupplyRes);
+
+  const userCLqdrBalances: any = [];
+  for (let i = 0; i < addresses.length - 1; i++) {
+    userCLqdrBalances[addresses[i]] = bn(0);
+  }
+
+  const clqdrMulti = new Multicaller(network, provider, contractAbi, { blockTag });
+  addresses.forEach((address) =>
+    clqdrMulti.call(address, options.address, 'balanceOf', [address])
+  );
+  const clqdrToken: Record<string, BigNumberish> = await clqdrMulti.execute();
+
+  Object.fromEntries(
+    Object.entries(clqdrToken).map(([address, balance]) => {
+      return addUserBalance(userCLqdrBalances, address, balance);
+    })
+  );
+
+  const userLpBalances: any = [];
+  for (let i = 0; i < addresses.length - 1; i++) {
+    userLpBalances[addresses[i]] = bn(0);
+  }
+
+  const multi = new Multicaller(network, provider, contractAbi, { blockTag });
+  addresses.forEach((address) =>
+    multi.call(address, lpAddress, 'balanceOf', [address])
+  );
+  const resultToken: Record<string, BigNumberish> = await multi.execute();
+
+  Object.fromEntries(
+    Object.entries(resultToken).map(([address, balance]) => {
+      return addUserBalance(userLpBalances, address, balance);
+    })
+  );
+
+  const multiLiquidMaster = new Multicaller(network, provider, contractAbi, {
+    blockTag
+  });
+
+  addresses.forEach((address) =>
+    multiLiquidMaster.call(address, liquidMasterAddress, 'userInfo', ['43', address])
+  );
+  const resultLiquidMaster: Record<string, BigNumberish> =
+    await multiLiquidMaster.execute();
+
+  Object.fromEntries(
+    Object.entries(resultLiquidMaster).map(([address, balance]) => {
+      return addUserBalance(userLpBalances, address, balance[0]);
+    })
+  );
+
+  const multiBeetsMaster = new Multicaller(network, provider, contractAbi, {
+    blockTag
+  });
+
+  addresses.forEach((address) =>
+    multiBeetsMaster.call(address, beetsMasterAddress, 'userInfo', ['69', address])
+  );
+  const resultBeetsMaster: Record<string, BigNumberish> =
+    await multiBeetsMaster.execute();
```

Comment:
It should not cross 5 request limit, https://github.com/snapshot-labs/snapshot-strategies#checklist-for-adding-a-new-strategy